### PR TITLE
IGNITE-20597 Support cache group filter for dumps

### PR DESCRIPTION
**Pull request by nizhikov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java [hunk 1]
```diff
@@ -2299,6 +2302,7 @@ public IgniteFutureImpl<Void> createSnapshot(
             }
 
             List<String> grps = (dump ? cctx.cache().cacheGroupDescriptors().values() : cctx.cache().persistentGroups()).stream()
+                .filter(g -> cacheGroupNames == null || cacheGroupNames.contains(g.cacheOrGroupName()))
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java:2305]: What if `cacheGroupNames` param contains non-existent cache group. Should we stop snapshot process? Or at least write a warning.

File: modules/core/src/main/java/org/apache/ignite/IgniteSnapshot.java [hunk 2]
```diff
@@ -69,9 +70,10 @@ public interface IgniteSnapshot {
      * In-memory cache groups also supported.
      *
      * @param name Dump name.
+     * @param cacheGroupNames Cache groups to include in snapshot or {@code null} to include all.
      * @return Future which will be completed when dump ends.
      */
-    public IgniteFuture<Void> createDump(String name);
+    public IgniteFuture<Void> createDump(String name, @Nullable Set<String> cacheGroupNames);
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/IgniteSnapshot.java:76]: Replace Set with Collection, as this style used in `restoreSnapshot` methods.

**Review by timoninmaxim — APPROVED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java (hunk 1 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java]: Warning added. Thanks.

File: modules/core/src/main/java/org/apache/ignite/IgniteSnapshot.java (hunk 2 above)
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/IgniteSnapshot.java]: Fixed.

**Pull request merged**